kale/loaddata/multi_domain.py

```diff
@@ -123,25 +125,26 @@ def __init__(
         self._source_by_split: Dict[str, torch.utils.data.Subset] = {}
         self._labeled_target_by_split = None
         self._target_by_split: Dict[str, torch.utils.data.Subset] = {}
+        self.class_ids = class_ids
 
     def is_semi_supervised(self):
         return self._n_fewshot is not None and self._n_fewshot > 0
 
     def prepare_data_loaders(self):
         logging.debug("Load source")
         (self._source_by_split["train"], self._source_by_split["valid"],) = self._source_access.get_train_val(
-            self._val_split_ratio
+            self._val_split_ratio, self.class_ids
         )
 
         logging.debug("Load target")
         (self._target_by_split["train"], self._target_by_split["valid"],) = self._target_access.get_train_val(
-            self._val_split_ratio
+            self._val_split_ratio, self.class_ids
         )
 
         logging.debug("Load source Test")
-        self._source_by_split["test"] = self._source_access.get_test()
+        self._source_by_split["test"] = self._source_access.get_test_class_subset(self.class_ids)
```

Comment:
I think the issue with this is that `get_train()` and `get_test()` are being redefined by its child classes. So, similar flags would have to be handled by each of the child classes separately. `get_train_val()` is not being re-defined, so I have added flags inside it.

Reply:
I think another option could be to have the child classes redefine a different function (maybe `_get_train()` private) and call this inside `get_train()`?

Reply:
> I think another option could be to have the child classes redefine a different function (maybe `_get_train()` private or `get_train_all()`) and call this inside `get_train()` with class_id flags?

It will be a good idea if feasible.